include/deal.II/base/symmetric_tensor.h

```diff
@@ -1215,17 +1215,15 @@ namespace internal
       {
       case 1:
         return data[0] * sdata[0];
-      case 2:
-        return (data[0] * sdata[0] +
-                data[1] * sdata[1] +
-                NumberType<Number>::value(2.0) * data[2] * sdata[2]);
```

Comment:
No, I did't do it because of vectorization here (I think that might first apply in 3D and even there I'm not sure because you'd need something that Intel calls horizontal additions) but rather because I don't think it's worthwhile to manually unroll loops any more and thus duplicate code. I might have helped 15 years ago when the initial tensor loops were written but today all sane compilers do in fact unroll and result in the exact same code as if it were written out (I looked at assembler code of gcc, clang and icc).